### PR TITLE
Verify command

### PR DESCRIPTION
**Pull request by Hoverbear**
Completes https://ferroussystems.clickup.com/t/86941z3t2

To test, run `criticalup verify` in an applicable folder. Mostly based off @amanjeev 's `install` code.

You can pass `-vvvvvv` (with however number of `v`s makes you happy) and see the tracing output of files being added to the validator.

### CODE REVIEW

File: crates/criticalup-cli/src/commands/verify.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    let mut integrity_verifier = IntegrityVerifier::new(keys);
+
+    let product_name = product.name();
+    let product_path = installation_dir.join(product.installation_id());
+    span.record(
+        "product_path",
+        tracing::field::display(product_path.display()),
+    );
+
+    tracing::info!("Verifying project `{product_name}`");
```
Comment (amanjeev) [inline crates/criticalup-cli/src/commands/verify.rs:76]:
```suggestion
    tracing::info!("Verifying product `{product_name}`");
```

We use `project` for user's project/criticalup.toml. I think you meant `product` here. Please correct me if I am mistaken.

File: crates/criticalup-cli/src/commands/verify.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                &tokio::fs::read(&entry.path()).await?,
+            );
+        }
+    }
+
+    integrity_verifier
+        .verify()
+        .map_err(Error::IntegrityErrorsWhileVerifying)?;
+
+    tracing::info!("Successfully verified `{product_name}`");
```
Comment (amanjeev) [inline crates/criticalup-cli/src/commands/verify.rs:95]:
```suggestion
    tracing::info!("Successfully verified '{product_name}'");
```

**Comment by amanjeev** [comment]
bors try

**Review by amanjeev — APPROVED** [review]
:heavy_check_mark: :heart:

**Comment by Hoverbear** [comment]
bors merge

**Pull request merged**